src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/PackageSourceTelemetry.cs

```diff
@@ -148,113 +206,77 @@ internal static TelemetryEvent ToTelemetry(Data data, string source, PackageSour
                 });
 
-            // source info
-            telemetry.AddPiiData(PropertyNames.Source.Url, source);
+            await AddSourcePropertiesAsync(telemetry, sourceRepository);
+            telemetry[PropertyNames.Duration.Total] = data.Resources.Values.Sum(r => r.duration.TotalMilliseconds);
+            telemetry[PropertyNames.Nupkgs.Copied] = data.NupkgCount;
+            telemetry[PropertyNames.Nupkgs.Bytes] = data.NupkgSize;
+            AddResourceProperties(telemetry, data.Resources);
 
-            if (packageSource == null)
+            if (data.Http.Requests > 0)
             {
-                packageSource = new PackageSource(source);
+                AddHttpProperties(telemetry, data.Http);
             }
 
-            if (packageSource.IsHttp)
-            {
-                telemetry[PropertyNames.Source.Type] = "http";
-                telemetry[PropertyNames.Source.Protocol] = TelemetryUtility.IsHttpV3(packageSource) ? 3 : packageSource.ProtocolVersion;
-            }
-            else
-            {
-                telemetry[PropertyNames.Source.Type] = "local";
-                telemetry[PropertyNames.Source.Protocol] = packageSource.ProtocolVersion;
-            }
+            return telemetry;
+        }
+
+        private static async Task AddSourcePropertiesAsync(TelemetryEvent telemetry, SourceRepository sourceRepository)
+        {
+            telemetry.AddPiiData(PropertyNames.Source.Url, sourceRepository.PackageSource.Source);
+
+            telemetry[PropertyNames.Source.Type] = await sourceRepository.GetFeedType(CancellationToken.None);
 
-            var msFeed = GetMsFeed(packageSource);
+            var msFeed = GetMsFeed(sourceRepository.PackageSource);
             if (msFeed != null)
             {
                 telemetry[PropertyNames.Source.MSFeed] = msFeed;
             }
+        }
 
-            AddResourceProperties(telemetry, data.Metadata, PropertyNames.Metadata);
-            AddResourceProperties(telemetry, data.Nupkg, PropertyNames.Nupkg);
-
-            ResourceData all = CalculateTotal(data.Metadata, data.Nupkg);
-            AddResourceProperties(telemetry, all, PropertyNames.All);
-
-            return telemetry;
+        private static void AddResourceProperties(TelemetryEvent telemetry, Dictionary<string, (int count, TimeSpan duration)> resources)
+        {
+            telemetry[PropertyNames.Resources.Calls] = resources.Values.Sum(r => r.count);
+            telemetry.ComplexData[PropertyNames.Resources.Details] = ToResourceDetailsTelemetry(resources);
         }
 
-        private static void AddResourceProperties(TelemetryEvent telemetry, ResourceData data, PackageSourceDiagnosticsPropertyNames.ResourcePropertyNames propertyNames)
+        private static void AddHttpProperties(TelemetryEvent telemetry, HttpData data)
         {
-            lock (data.Lock)
+            telemetry[PropertyNames.Http.Requests] = data.Requests;
```

Comment:
Why remove the lock here? It's not possible for multiple threads to hit this at the same time?

Reply:
Previous code had two data objects per source, after the refactor there is one. It was intended that the `ToTelemetry` method gets the lock, but there was a bug which will be fixed in the next push.